See https://github.com/matplotlib/matplotlib/pull/14582

commit 4b957fde34c869e40d010d2ec871d3d219e59ee2
Author: Antony Lee <anntzer.lee@gmail.com>
Date:   Fri Jul 20 19:04:25 2018 +0200

    Backport PR #11722: Remove unnecessary hacks from setup.py.
    
    - There are no symlinks in the mpl source tree anymore.
    - MANIFEST was removed from the root directory in 365a0a2f (11 years
      ago).

diff --git a/setup.py b/setup.py
index 41037e9..b045f89 100644
--- a/setup.py
+++ b/setup.py
@@ -11,23 +11,6 @@ from setuptools.command.build_ext import build_ext as BuildExtCommand
 
 import sys
 
-# distutils is breaking our sdists for files in symlinked dirs.
-# distutils will copy if os.link is not available, so this is a hack
-# to force copying
-import os
-try:
-    del os.link
-except AttributeError:
-    pass
-
-# This 'if' statement is needed to prevent spawning infinite processes
-# on Windows
-if __name__ == '__main__':
-    # BEFORE importing distutils, remove MANIFEST. distutils doesn't properly
-    # update it when the contents of directories change.
-    if os.path.exists('MANIFEST'):
-        os.remove('MANIFEST')
-
 from io import BytesIO
 import os
 from string import Template
